src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SSRP.cs

```diff
@@ -116,6 +119,37 @@ private static byte[] CreateDacPortInfoRequest(string instanceName)
         }
 
         /// <summary>
+        /// Returns null-terminated string for given string.
+        /// </summary>
+        /// <param name="str">string to be null-terminated</param>
+        /// <returns>null-terminated string for given string</returns>
+        private static string GetNullTerminatedString(string str)
+        {
+            Debug.Assert(str != null, "str should not be null");
+
+            string result = null;
+            int nullCharIndex = str.IndexOf(NullChar);
+
+            // str does not have '\0'
+            if (nullCharIndex < 0)
```

Comment:
Are these checks necessary? Is there any scenario, where you can get a Null-Terminated string or a Null-Terminated character in the middle of the string?